lib/spack/spack/binary_distribution.py

```diff
@@ -873,15 +873,22 @@ def build_tarball(spec, outdir, force=False, rel=False, unsigned=False,
     spackfile_path = os.path.join(
         cache_prefix, tarball_path_name(spec, '.spack'))
 
-    remote_spackfile_path = url_util.join(
+    outdir = mirror.push_url
+    fetch_spackfile_path = url_util.join(
+        mirror.fetch_url, os.path.relpath(spackfile_path, tmpdir))
+    push_spackfile_path = url_util.join(
         outdir, os.path.relpath(spackfile_path, tmpdir))
 
     mkdirp(tarfile_dir)
-    if web_util.url_exists(remote_spackfile_path):
+    if web_util.url_exists(push_spackfile_path):
         if force:
-            web_util.remove_url(remote_spackfile_path)
+            web_util.remove_url(push_spackfile_path)
         else:
-            raise NoOverwriteException(url_util.format(remote_spackfile_path))
+            raise NoOverwriteException(url_util.format(push_spackfile_path))
+
+    if web_util.url_exists(fetch_spackfile_path):
+        if not force:
+            raise NoOverwriteException(url_util.format(fetch_spackfile_path))
```

Comment:
I'm a bit confused by this and it may just be variable naming.  Why do we end up pushing to a 'fetch' URL?  Seems to me that it would be a read-only kind of thing.

Reply:
It is RO, and we're not pushing to it. We do check for existence at the fetch url, and only creating the binary cache package if it doesn't exist there.